sphere_tesselate.f90

```diff
@@ -899,6 +918,436 @@ function tolower(a) result(b)
+   inquire(file=dir, exist=dir_exists)
+   if (.not.dir_exists) then
+      write(0,'(a)') 'st_save_neighbours_cache: Error: Directory "' // trim(dir) // &
+         '" does not exist'
+      error stop
+   endif
+   write(file,'("ico_pole_",i0.1,".nbrs")') lnbrs%level
+   file = trim(dir) // '/' // trim(file)
+   call st_save_neighbours(lnbrs, file)
+end subroutine st_save_neighbours_cache
+!-------------------------------------------------------------------------------
+
+!===============================================================================
+subroutine st_load_neighbours_cache(level, lnbrs, exists)
+!===============================================================================
+!  Load a tesselation from the default cache location.  The level value must
+!  be filled with the desired tesselation level.  At the moment, only polar,
+!  icosahedral tesselations are supported.
+!  Optionally, the exists argument specified whether the cache file exists.
+!  If this is supplied, the routine will not exit with an error when the file
+!  does not exist.
+   integer, intent(in) :: level
+   type(list_neighbours), intent(out) :: lnbrs
+   logical, optional, intent(out) :: exists
+   logical :: file_exists
+   character(len=250) :: dir, file
+
+   call get_environment_variable(st_cache_env_var, dir)
+   if (dir == '') dir = st_cache_dir
+   if (level < 0 .or. level > 15) then
+      write(0,'(a,i0.1)') 'st_load_neighbours_cache: Error: tesselation level is ', level
+      error stop
+   endif
+   write(file,'("ico_pole_",i0.1,".nbrs")') level
+   file = trim(dir) // '/' // trim(file)
+   inquire(file=file, exist=file_exists)
+   if (present(exists)) then
+      exists = file_exists
+      if (.not.file_exists) return
+   else
+      if (.not.file_exists) then
+         write(0,'(a)') 'st_load_neighbours_cache: Error: Cache file "' // trim(file) &
+            // '" does not exist'
+         error stop
+      endif
+   endif
+   call st_load_neighbours(file, lnbrs)
+end subroutine st_load_neighbours_cache
+!-------------------------------------------------------------------------------
+
+!===============================================================================
+subroutine st_save_neighbours(lnbrs, file, ascii)
+!===============================================================================
+!  Save neighbours to an output file.  This can be ASCII or Fortran binary;
+!  use ascii=.true. for the former.  These can then be re-read in later.
+!  Binary format is the default
+   type(list_neighbours), intent(in)        :: lnbrs 
+   character(len=*), intent(in) :: file
+   logical, optional, intent(in) :: ascii
+   logical :: ascii_in
+   integer :: i, iostat
+   character(len=30) :: string
+
+   ascii_in = .false.
+   if (present(ascii)) ascii_in = ascii
+
+
+   ! ASCII file
+   if (ascii_in) then
+      open(lu, file=file, iostat=iostat)
+      call check_open
+      write(lu,'(a,i0.1)') '# sphere_tesselate version ', st_version
```

Comment:
I realise this is the same header file but the information needed is basically the same and is essentially invisible to the end user anyway.  The neighbours cache file is distinguished from the tesselation files by the extension.  The code writes neighbours cache files with a ".nbrs" extension (compared with the ".tess" extension otherwise used.)